src/util/tokenpipe.h

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_TOKENPIPE_H
+#define BITCOIN_UTIL_TOKENPIPE_H
+
+#ifndef WIN32
+
+#include <cstdint>
+#include <optional>
+
+/** One end of a token pipe. */
+class TokenPipeEnd {
+private:
+    int m_fd = -1;
+public:
```

Comment:
style-nit, suggested by [clang-format-diff.py](https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy):
```suggestion
class TokenPipeEnd
{
private:
    int m_fd = -1;

public:
```